### PR TITLE
[Followup] fix problems with creating example solutions and bugs in TextAssessmentResource

### PR DESCRIPTION
**Pull request by PatZintel**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database 

### Motivation and Context

follow-up for the refactoring of the Result- and TextAssessmentResource

### Description

added Tests

### Steps for Testing

check the console for errors such as badRequest related to the Endpoints in TextAssessmentResource and ResultResource while you:
1. create a text exercise
2. participate in it twice
3. assess one participation
4. start assessing the second one, dont submit and go back to the assessment dashboard, then continue assessing the exercise
5. create a n example submission with assessment and save it
6. assess a text exercises in an exam (maybe use the test run feature here, makes it faster to test instead of conducting a whole 
7.  exam ^^)

code review for test additions

### Test Coverage

ResultResource: 93%
TextAssessmentResource: 91%

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/ExampleSubmissionIntegrationTest.java [hunk 1]
```diff
@@ -205,4 +205,48 @@ public void createExampleTextAssessment() throws Exception {
...
+    public void createExampleTextAssessment_notExistentId() throws Exception {
+        ExampleSubmission storedExampleSubmission = database.addExampleSubmission(database.generateExampleSubmission("Text. Submission.", textExercise, true));
+        database.addResultToSubmission(storedExampleSubmission.getSubmission(), AssessmentType.MANUAL);
+        final Result exampleResult = request.get("/api/exercises/" + textExercise.getId() + "/submissions/" + storedExampleSubmission.getSubmission().getId() + "/example-result",
+                HttpStatus.OK, Result.class);
+        final Set<TextBlock> blocks = ((TextSubmission) exampleResult.getSubmission()).getBlocks();
+        assertThat(blocks).hasSize(2);
+        List<Feedback> feedbacks = new ArrayList<>();
+        final Iterator<TextBlock> textBlockIterator = blocks.iterator();
+        feedbacks.add(new Feedback().credits(80.00).type(FeedbackType.MANUAL).detailText("nice submission 1").reference(textBlockIterator.next().getId()));
```
Comment (MerlinMehmed) [inline src/test/java/de/tum/in/www1/artemis/ExampleSubmissionIntegrationTest.java:220]: Maybe you can use the ModelFactory (i.e. generateManualFeedback()) here and in the next test to generate feedbacks in order to remove duplicates.

**Review by MerlinMehmed — CHANGES_REQUESTED** [review]
Additionally, it would also be good if you can fix the Codacy issues. 

I couldn't test the integration of the client with all of the endpoints because of the problem in participations/{participationId}/submissions/{submissionId}/for-text-assessment so I will test them after this is fixed.

File: src/main/webapp/app/exercises/text/assess/text-submission-assessment.route.ts [hunk 2]
```diff
@@ -101,7 +101,7 @@ export const textSubmissionAssessmentRoutes: Routes = [
         canActivate: [UserRouteAccessService],
     },
     {
-        path: 'submissions/:submissionId/assessment',
+        path: 'participations/:participationId/submissions/:submissionId/for-text-assessment',
```
Comment (krusche) [inline src/main/webapp/app/exercises/text/assess/text-submission-assessment.route.ts:104]: if you change those paths, you probably also need to change the component that load the corresponding pages

**Review by Olimbia23 — COMMENTED** [review]
Creation of text and modelling- example submissions seems to be working. 
The creation of Example Assessment also works. @PatZintel

**Review by Olimbia23 — DISMISSED** [review]
re-tested on TS1 the creation of Text-/Model Example Submissions. 
As well as for exams. 
It works as it should. 👍 @PatZintel

**Review by imstreetspirit — CHANGES_REQUESTED** [review]
tested on ts3, seems good for normal exercise/exam mode. Only thing, this Save button did not work for me while assessing the text submissions. tried for two submissions, it throws the same error. Submit works as expected.

**Review by argertboja — COMMENTED** [review]
Works as expected on TS1!

**Review by imstreetspirit — APPROVED** [review]
saving works now.. tested on ts3 again.

**Pull request merged**